superset/assets/src/dashboard/util/getPublishSubscriberMap.js

```diff
@@ -20,135 +20,148 @@ import * as _ from 'lodash';
+  return (slice.formData.linked_slice && slice.formData.linked_slice.length > 0)
 }
 
 function getSubscriberMap(slices, publishers) {
-    var subs;
-    slices.forEach(element => {
-        if (isSubscriber(element)) {
-            if (!subs) {
-                subs = {};
-            }
-            subs[element.id] = createSubscriberDataFor(element, publishers)
-        }
-    });
-    return subs;
+  var subs;
+  slices.forEach(element => {
+    if (isSubscriber(element)) {
+      if (!subs) {
+        subs = {};
+      }
+      subs[element.id] = createSubscriberDataFor(element, publishers)
+    }
+  });
+  return subs;
 }
 
 export default function getPublishSubscriberMap(slices) {
-    slices = updateSlices(slices);
-    let publishers = getPublisherMap(slices);
-    let subscribers = publishers ? getSubscriberMap(slices, publishers) : undefined;
-    return {
-        publishers: publishers,
-        subscribers: subscribers
-    };
+  slices = updateSlices(slices);
+  let publishers = getPublisherMap(slices);
+  let subscribers = publishers ? getSubscriberMap(slices, publishers) : undefined;
+  return {
+    publishers: publishers,
+    subscribers: subscribers
+  };
 }
 
 function updateSlices(slices) {
-    let updatedSlices = _.clone(slices);
-    updatedSlices.forEach(slice => {
-        let linkedSlices = getLinkedSlicesFromSubscriberLayer(slice.formData.subscriber_layers);
-        slice.formData.linked_slice = linkedSlices ? linkedSlices : slice.formData.linked_slice;
-        slice.formData.actions = ['APPLY_FILTER'];
+  let updatedSlices = _.clone(slices);
+  updatedSlices.forEach(slice => {
+    const linkedSlices = getLinkedSlicesFromSubscriberLayer(slice.formData.subscriber_layers);
+    slice.formData.linked_slice = linkedSlices ? linkedSlices : slice.formData.linked_slice;
+    slice.formData.actions = getUniqueActionsForSlice(slice.formData.subscriber_layers);
+    slice.formData.useAsModal = slice.formData.actions && slice.formData.actions.indexOf('USE_AS_MODAL') >= 0 ? true : false;
+  });
+
+  return updatedSlices;
+}
+
+function getUniqueActionsForSlice(subscriberLayers) {
+  let actions = [];
+
+  if (subscriberLayers) {
+    subscriberLayers.forEach(element => {
+      actions = _.union(actions, element.actions);
     });
+  }
 
-    return updatedSlices;
+  return actions;
 }
 
 function getLinkedSlicesFromSubscriberLayer(subscriberLayer) {
-    let linkedSlices;
-    if (subscriberLayer) {
-        linkedSlices = [];
-        subscriberLayer.forEach(element => {
-            element.linked_slice.forEach(item => {
-                linkedSlices.push(item);
-            });
-        });
-    }
+  let linkedSlices;
+  if (subscriberLayer) {
+    linkedSlices = [];
+    subscriberLayer.forEach(element => {
+      element.linked_slice.forEach(item => {
+        linkedSlices.push(item);
+      });
+    });
+  }
 
-    return linkedSlices;
+  return linkedSlices;
 }
```

Comment:
lot of unwanted changes can we avoid it and re-format so that only required changes are visible